crates/sui/src/upgrade_compatibility.rs

```diff
@@ -813,6 +821,47 @@ fn module_compatibility_error_diag(
     Ok(diags)
 }
 
+const PACKAGE_TABLE: &str = "[package]";
+fn missing_module_diag(
+    module_name: &Identifier,
+    package_path: &PathBuf,
+) -> Result<Diagnostics, Error> {
+    let mut diags = Diagnostics::new();
+
+    // read Move.toml to get the hash and first line start and end
+    let toml_path = package_path.join("Move.toml");
+    let toml_str = fs::read_to_string(&toml_path).context("Unable to read Move.toml")?;
+    let hash = FileHash::new(&toml_str);
+
+    let start: usize = toml_str
+        .find(PACKAGE_TABLE)
+        .context("Malformed Move.toml")?;
+    // default to the end of the package table definition
+    let mut end = start + PACKAGE_TABLE.len();
+
+    // get the third newline after the start of the package table declaration if it exists
+    toml_str
+        .match_indices('\n')
+        .filter(|(idx, _)| *idx > start)
+        .take(3)
+        .last()
+        .map(|(idx, _)| end = idx);
```

Comment:
Slightly more idiomatic:

```suggestion
    // Include up to three newlines after the start of the package table declaration, or
    // default to the end of the `[package]` key.
    let end = toml_str[start..]
        .match_indices('\n')
        .take(3)
        .last()
        .unwrap_or(start + PACKAGE_TABLE.len());
```